packages/vue/src/components/confirm-sign-in.vue

```diff
@@ -1,66 +1,61 @@
 <script setup lang="ts">
+import { computed, toRefs, useAttrs } from 'vue';
 import {
   authenticatorTextUtil,
-  getActorState,
   getFormDataFromEvent,
-  SignInState,
   translate,
 } from '@aws-amplify/ui';
-import { computed, ComputedRef, useAttrs } from 'vue';
 
-import { useAuth, useAuthenticator } from '../composables/useAuth';
+import { useAuthenticator } from '../composables/useAuth';
+import { UseAuthenticator } from '../types';
 import BaseFormFields from './primitives/base-form-fields.vue';
 
+/** @deprecated Component events are deprecated and not maintained. */
 const emit = defineEmits(['confirmSignInSubmit', 'backToSignInClicked']);
 const attrs = useAttrs();
 
-const { state, send } = useAuth();
+// `useAuthenticator` is casted for temporary type safety on this file.
+const props = useAuthenticator() as UseAuthenticator;
```

Comment:
Can we achieve the same with type annotation?

Reply:
and other places.

Reply:
[44667e1](https://github.com/aws-amplify/amplify-ui/pull/4202/commits/44667e1b5ade482c247e86d86c485a2ffcf27763) Done! And I'll make a follow up PR for making this change to other subcomponents that are outside this PR.